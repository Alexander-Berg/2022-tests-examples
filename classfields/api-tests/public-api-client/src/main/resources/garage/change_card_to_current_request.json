{
  "cardType": "CURRENT_CAR",
  "exCarAdditionalInfo": {
    "reason": "SOLD"
  }
}