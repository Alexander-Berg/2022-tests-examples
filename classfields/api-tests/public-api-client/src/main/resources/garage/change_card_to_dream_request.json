{
  "cardType": "DREAM_CAR",
  "exCarAdditionalInfo": {
    "reason": "SOLD"
  }
}