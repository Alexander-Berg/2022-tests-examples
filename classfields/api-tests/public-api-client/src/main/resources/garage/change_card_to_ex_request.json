{
  "cardType": "EX_CAR",
  "exCarAdditionalInfo": {
    "reason": "SOLD"
  }
}