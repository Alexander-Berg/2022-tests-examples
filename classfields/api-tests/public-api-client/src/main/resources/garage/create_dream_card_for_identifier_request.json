{
  "card_type": "DREAM_CAR",
  "registration_region_id": 203
}