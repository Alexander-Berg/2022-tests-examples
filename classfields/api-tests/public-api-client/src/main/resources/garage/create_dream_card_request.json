{
  "added_manually": true,
  "card": {
    "vehicle_info": {
      "car_info": {
        "mark": "BMW",
        "model": "M5",
        "super_gen_id": "3722944"
      },
      "color": {
        "id": "0000CC"
      }
    },
    "card_type_info": {
      "card_type": "DREAM_CAR"
    }
  },
  "geo_radius": 200,
  "geo_id": [
    213
  ]
}