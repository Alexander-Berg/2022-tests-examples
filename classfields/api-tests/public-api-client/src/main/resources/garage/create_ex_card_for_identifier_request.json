{
  "card_type": "EX_CAR",
  "registration_region_id": 203
}