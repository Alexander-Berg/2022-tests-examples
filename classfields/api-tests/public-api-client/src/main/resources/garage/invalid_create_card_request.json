{
  "card": {
    "vehicle_info": {
      "car_info": {
        "body_type": "ALLROAD_5_DOORS",
        "engine_type": "GASOLINE",
        "transmission": "VARIATOR",
        "drive": "ALL_WHEEL_DRIVE",
        "mark": "MITSUBISHI",
        "model": "OUTLANDER",
        "super_gen_id": "20526645",
        "configuration_id": "20526649",
        "complectation_id": "20526825",
        "steering_wheel": "LEFT"
      },
      "documents": {
        "owners_number": 20,
        "year": 1900,
        "vin": "X9FLXXEEBLES67719"
      }
    }
  }
}