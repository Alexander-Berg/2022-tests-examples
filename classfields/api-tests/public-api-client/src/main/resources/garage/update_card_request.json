{
  "card": {
    "vehicle_info": {
      "car_info": {
        "body_type": "SEDAN",
        "engine_type": "GASOLINE",
        "transmission": "MECHANICAL",
        "drive": "FORWARD_CONTROL",
        "mark": "PEUGEOT",
        "model": "408",
        "super_gen_id": "8350849",
        "configuration_id": "8350853",
        "tech_param_id": "20089520",
        "horse_power": 110
      },
      "documents": {
        "owners_number": 3,
        "year": 2013,
        "vin": "Z8T4DNFUCDM014995"
      }
    }
  }
}