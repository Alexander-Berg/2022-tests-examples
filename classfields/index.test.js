const cardImageGallery = require('./index');
const getValidInteriorPanoramaData = require('./getValidInteriorPanoramaData');

const cloneOfferWithHelpers = require('autoru-frontend/mockData/state/helpers/offer/cloneOfferWithHelpers');

let offer;
beforeEach(() => {
    offer = {
        state: {
            image_urls: [
                { sizes: { '1200x900n': 'img1' } },
                { sizes: { '1200x900n': 'img2' } },
                { sizes: { '1200x900n': 'img3' } },
            ],
        },
    };
});

describe('порядок galleryItems', () => {
    it('должен расположить фотки в том же порядке', () => {
        expect(cardImageGallery(offer).galleryItems).toEqual([
            { sizes: { '1200x900n': 'img1' }, type: 'IMAGE' },
            { sizes: { '1200x900n': 'img2' }, type: 'IMAGE' },
            { sizes: { '1200x900n': 'img3' }, type: 'IMAGE' },
        ]);
    });

    describe('есть панорама', () => {
        beforeEach(() => {
            offer = cloneOfferWithHelpers(offer).withPanoramaExterior();
        });

        it('должен расположить: внешняя панорама, внутренняя панорама, видео, фотки', () => {
            offer = offer.withPanoramaInterior().withYoutubeVideo().value();

            expect(cardImageGallery(offer).galleryItems).toEqual([
                { ...offer.state.external_panorama.published, type: 'PANORAMA_EXTERIOR' },
                getValidInteriorPanoramaData(offer),
                {
                    type: 'VIDEO',
                    provider: 'YOUTUBE',
                    original: '//i.ytimg.com/vi/ytvideo/hqdefault.jpg',
                    thumbnail: '//i.ytimg.com/vi/ytvideo/mqdefault.jpg',
                    video: '//www.youtube.com/embed/ytvideo?enablejsapi=1',
                },
                { sizes: { '1200x900n': 'img1' }, type: 'IMAGE' },
                { sizes: { '1200x900n': 'img2' }, type: 'IMAGE' },
                { sizes: { '1200x900n': 'img3' }, type: 'IMAGE' },
            ]);
        });

        it('должен расположить: панорама, спинкар, видео, фотки', () => {
            offer = offer.withSpincar().withYoutubeVideo().value();

            expect(cardImageGallery(offer).galleryItems).toEqual([
                { ...offer.state.external_panorama.published, type: 'PANORAMA_EXTERIOR' },
                { sizes: { '1200x900n': 'img2' }, spincar: 'spincar_exterior_url', type: 'SPINCAR' },
                {
                    type: 'VIDEO',
                    provider: 'YOUTUBE',
                    original: '//i.ytimg.com/vi/ytvideo/hqdefault.jpg',
                    thumbnail: '//i.ytimg.com/vi/ytvideo/mqdefault.jpg',
                    video: '//www.youtube.com/embed/ytvideo?enablejsapi=1',
                },
                { sizes: { '1200x900n': 'img1' }, type: 'IMAGE' },
                { sizes: { '1200x900n': 'img2' }, type: 'IMAGE' },
                { sizes: { '1200x900n': 'img3' }, type: 'IMAGE' },
            ]);
        });
    });

    describe('нет панорамы', () => {
        beforeEach(() => {
            offer = cloneOfferWithHelpers(offer);
        });

        it('должен расположить: фотка, внутренняя панорама, видео, фотки', () => {
            offer = offer.withPanoramaInterior().withYoutubeVideo().value();

            expect(cardImageGallery(offer).galleryItems).toEqual([
                { sizes: { '1200x900n': 'img1' }, type: 'IMAGE' },
                getValidInteriorPanoramaData(offer),
                {
                    type: 'VIDEO',
                    provider: 'YOUTUBE',
                    original: '//i.ytimg.com/vi/ytvideo/hqdefault.jpg',
                    thumbnail: '//i.ytimg.com/vi/ytvideo/mqdefault.jpg',
                    video: '//www.youtube.com/embed/ytvideo?enablejsapi=1',
                },
                { sizes: { '1200x900n': 'img2' }, type: 'IMAGE' },
                { sizes: { '1200x900n': 'img3' }, type: 'IMAGE' },
            ]);
        });

        it('должен расположить: фотка, видео, спинкар, фотки', () => {
            offer = offer.withSpincar().withYoutubeVideo().value();

            expect(cardImageGallery(offer).galleryItems).toEqual([
                { sizes: { '1200x900n': 'img1' }, type: 'IMAGE' },
                {
                    type: 'VIDEO',
                    provider: 'YOUTUBE',
                    original: '//i.ytimg.com/vi/ytvideo/hqdefault.jpg',
                    thumbnail: '//i.ytimg.com/vi/ytvideo/mqdefault.jpg',
                    video: '//www.youtube.com/embed/ytvideo?enablejsapi=1',
                },
                { sizes: { '1200x900n': 'img2' }, spincar: 'spincar_exterior_url', type: 'SPINCAR' },
                { sizes: { '1200x900n': 'img2' }, type: 'IMAGE' },
                { sizes: { '1200x900n': 'img3' }, type: 'IMAGE' },
            ]);
        });
    });

    describe('бейджи', () => {
        const offerWithOrigImg = {
            state: {
                image_urls: [
                    { sizes: { '1200x900n': 'img1' } },
                    { sizes: { '1200x900n': 'img2' }, is_hd: true },
                    { sizes: { '1200x900n': 'img3' } },
                ],
            },
        };

        it('должен добавить HD', () => {
            offer = cloneOfferWithHelpers(offerWithOrigImg).withStatus('ACTIVE').value();
            offer.documents = { vin_resolution: 'OK' };

            expect(cardImageGallery(offer).badges.hd).toBeTruthy();
        });

        it('не должен добавить HD, если у оффера нет отчёта', () => {
            offer = cloneOfferWithHelpers(offerWithOrigImg).value();

            expect(cardImageGallery(offer).badges.hd).toBeFalsy();
        });

        it('не должен добавить HD, если у оффера нет hd-фоток', () => {
            offer = cloneOfferWithHelpers(offer).withTags([ 'vin_offers_history' ]).value();

            expect(cardImageGallery(offer).badges.hd).toBeFalsy();
        });
    });
});
