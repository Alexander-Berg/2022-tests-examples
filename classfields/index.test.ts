import { InsuranceType, InsuranceStatus } from '@vertis/schema-registry/ts-types-snake/auto/api/vin/vin_report_model';
import { InsuranceSource } from '@vertis/schema-registry/ts-types-snake/auto/api/vin/garage/garage_api_model';

import getFormattedMoscowDate from 'auto-core/lib/util/time/getFormattedMoscowDate';

import { nbsp } from 'auto-core/react/lib/html-entities';

import dayjs from 'auto-core/dayjs';

import { getInsuranceData, getDaysTo } from './index';

Date.now = jest.fn(() => 1633640055424);

it('getDaysTo возвращает коректное количество дней', () => {
    const nowM10d = dayjs().add(-10, 'd');
    const nowP3d = dayjs().add(3 * 24 + 1, 'h');
    const result = getDaysTo({ from: nowM10d.toISOString(), to: nowP3d.toISOString() });

    expect(result).toBe('3 дня');
});

it('getDaysTo возвращает 5 дней', () => {
    const nowM10d = dayjs().add(-10, 'd');
    const nowP5d = dayjs().add(5 * 24 + 1, 'h');
    const result = getDaysTo({
        from: nowM10d.toISOString(),
        to: nowP5d.toISOString(),
    });

    expect(result).toBe('5 дней');
});

it('getDaysTo возвращает истёк', () => {
    const nowM1D = dayjs().add(-1, 'd');
    const nowM3d = dayjs().add(-365, 'd');
    const result = getDaysTo({ from: nowM3d.toISOString(), to: nowM1D.toISOString() });

    expect(result).toBe('истёк');
});

it('getDaysTo не возвращает 0', () => {
    const now = dayjs();
    const nowM3d = dayjs().add(-365, 'd');
    const result = getDaysTo({ from: nowM3d.toISOString(), to: now.toISOString() });

    expect(result).toBe('1 день');
});

it('getInsuranceData возвращает минимум 2 поля: "Начало действия" и "Серия и номер"', () => {
    const result = getInsuranceData({
        insurance_type: InsuranceType.OSAGO,
        is_actual: true,
        number: '111',
        serial: '',
        status: InsuranceStatus.ACTIVE,
        source: InsuranceSource.UNKNOWN_SOURCE,
    });

    expect(result).toMatchSnapshot();
});

it('getInsuranceData возвращает "Окончание действия"', () => {
    const now = dayjs();
    const nowM1Y = dayjs().add(-365, 'd');
    const nowM1D = dayjs().add(-1, 'd');
    const props: { [key: string]: string | boolean | undefined } = {};

    getInsuranceData({
        serial: '',
        number: '111',
        insurance_type: InsuranceType.OSAGO,
        status: InsuranceStatus.ACTIVE,
        from: nowM1Y.toISOString(),
        is_actual: true,
        source: InsuranceSource.UNKNOWN_SOURCE,
    }).forEach(x => (props[x.key] = x.value));

    expect(props.insuranceTo).toBe(getFormattedMoscowDate(now).replace(/\s/g, nbsp));
    expect(props.serialNumber).toBe('111');

    getInsuranceData({
        serial: 'XXX',
        number: '111',
        insurance_type: InsuranceType.KASKO,
        status: InsuranceStatus.ACTIVE,
        from: nowM1Y.toISOString(),
        to: nowM1D.toISOString(),
        is_actual: true,
        source: InsuranceSource.UNKNOWN_SOURCE,
    }).forEach(x => (props[x.key] = x.value));

    expect(props.insuranceTo).toBe(getFormattedMoscowDate(nowM1D).replace(/\s/g, nbsp));
    expect(props.insuranceFrom).toBe(getFormattedMoscowDate(nowM1Y).replace(/\s/g, nbsp));
    expect(props.serialNumber).toBe('XXX 111');

});
