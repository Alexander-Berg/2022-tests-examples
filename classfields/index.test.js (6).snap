// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен вернуть правильное описание параметров поиска для CARS {"catalog_equipment":["led-lights,xenon","rain-sensor"],"section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "catalog_equipment",
      "label": undefined,
      "val": "название led-lights",
    },
    Object {
      "key": "catalog_equipment",
      "label": undefined,
      "val": "название xenon",
    },
    Object {
      "key": "catalog_equipment",
      "label": undefined,
      "val": "название rain-sensor",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"currency":"USD","price_from":2000,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "price_from",
      "label": "Цена",
      "val": "от 2 000 $",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"displacement_from":600,"displacement_to":1000,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "displacement",
      "label": "Объем",
      "val": "0.6 — 1.0 л",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"displacement_from":600,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "displacement_from",
      "label": "Объем",
      "val": "от 0.6 л",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"displacement_to":1000,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "displacement_to",
      "label": "Объем",
      "val": "до 1.0 л",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"engine_group":["ELECTRO","TURBO","ATMO"],"section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "engine_group",
      "label": "Двигатель",
      "val": "Электро, Турбированный, Атмосферный",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"has_image":false,"has_history":true,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "has_image",
      "label": "Только с фото",
      "val": "Неважно",
    },
    Object {
      "val": "С историей",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"has_image":true,"has_history":true,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "val": "Только с фото",
    },
    Object {
      "val": "С историей",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"price_from":2000,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "price_from",
      "label": "Цена",
      "val": "от 2 000 ₽",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"search_tag":["handling","economical"],"section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "search_tag",
      "label": undefined,
      "val": "Хорошая управляемость",
    },
    Object {
      "key": "search_tag",
      "label": undefined,
      "val": "Экономичный",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"year_to":2017,"year_from":2019,"displacement_from":250,"displacement_to":1000,"section":"used","has_image":false,"has_history":true} 1`] = `
Object {
  "moreCount": 2,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "year",
      "label": "Год",
      "val": "2019 — 2017",
    },
    Object {
      "key": "displacement",
      "label": "Объем",
      "val": "0.3 — 1.0 л",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для CARS {"year_to":2019,"year_from":2019,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "year",
      "label": "Год",
      "val": "2019",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для MOTO {"atv_type":["UTILITARIAN","AMPHIBIAN","TOURIST","CHILDISH","BUGGI","SPORTS"],"moto_type":"ATV","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "atv_type",
      "label": "Тип вездехода",
      "val": "Утилитарный, Амфибия, Туристический, Детский, Багги, Спортивный",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для MOTO {"displacement_from":250,"displacement_to":1000,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "displacement_moto",
      "label": "Объем",
      "val": "250 — 1 000 см³",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для MOTO {"section":"all","moto_type":"MOTORCYCLE","engine_type":["INJECTOR","TURBO"]} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "engine_type",
      "label": "Двигатель",
      "val": "Инжектор, Бензин турбонаддув",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для MOTO {"section":"all","moto_type":"MOTORCYCLE","transmission":["TRANSMISSION_6","AUTOMATIC","AUTOMATIC_2_SPEED","ROBOTIC"]} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "transmission",
      "label": "Коробка",
      "val": "6 передач, АКПП, 2-скоростной автомат, Роботизированная",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"agricultural_type":["SEEDER"],"category":"TRUCKS","operating_hours_from":1,"operating_hours_to":10,"trucks_category":"AGRICULTURAL","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "operating_hours",
      "label": "Моточасы",
      "val": "1 — 10",
    },
    Object {
      "key": "agricultural_type",
      "label": "Тип техники",
      "val": "Сеялка",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"autoloader_type":["GRAB_LOADERS"],"category":"TRUCKS","trucks_category":"AUTOLOADER","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "autoloader_type",
      "label": "Тип автопогрузчика",
      "val": "Грейферный погрузчик",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"bucket_volume_from":1,"bucket_volume_to":20,"trucks_category":"DREDGE","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "bucket_volume",
      "label": "Объём ковша",
      "val": "1 — 20 м³",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"bulldozer_type":["CRAWLERS_BULLDOZER"],"category":"TRUCKS","traction_class":["TRACTION_10","TRACTION_6"],"trucks_category":"BULLDOZERS","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "bulldozer_type",
      "label": "Тип бульдозера",
      "val": "Гусеничный бульдозер",
    },
    Object {
      "key": "traction_class",
      "label": "Тяговый класс",
      "val": "10, 6",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"category":"TRUCKS","crane_radius_from":1,"trucks_category":"CRANE","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "crane_radius_from",
      "label": "Вылет стрелы",
      "val": "от 1 м",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"category":"TRUCKS","engine_type":["DIESEL"],"municipal_type":["WATERING_MACHINE"],"trucks_category":"MUNICIPAL","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "engine_type",
      "label": "Двигатель",
      "val": "Дизель",
    },
    Object {
      "key": "municipal_type",
      "label": "Тип техники",
      "val": "Поливомоечная машина",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"category":"TRUCKS","engine_type":["GASOLINE"],"trucks_category":"TRUCK","section":"all","suspension_chassis":["SPRING_PNEUMO"],"wheel_drive":["WD_6x4"]} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "engine_type",
      "label": "Двигатель",
      "val": "Бензин",
    },
    Object {
      "key": "suspension_chassis",
      "label": "Подвеска шасси",
      "val": "Рессора-пневмо",
    },
    Object {
      "key": "wheel_drive",
      "label": "Колёсная формула",
      "val": "6x4",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"category":"TRUCKS","euro_class":["EURO_2","EURO_GREEN"],"trucks_category":"TRUCK","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "euro_class",
      "label": "Класс выхлопов EURO",
      "val": "2, Зеленый сертификат",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"category":"TRUCKS","haggle":"HAGGLE_POSSIBLE","section":"all","trucks_category":"TRUCK"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "val": "Возможен торг",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"category":"TRUCKS","seats_from":1,"seats_to":10,"trucks_category":"LCV","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "seats",
      "label": "К-во мест",
      "val": "1 — 10",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"displacement_from":1200,"displacement_to":2600,"section":"used"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "section",
      "val": "С пробегом",
    },
    Object {
      "key": "displacement",
      "label": "Объем",
      "val": "1.2 — 2.6 л",
    },
  ],
}
`;

exports[`должен вернуть правильное описание параметров поиска для TRUCKS {"dredge_type":["CRAWLER_EXCAVATOR","TRENCH_EXCAVATOR"],"price_from":1,"price_to":2,"trucks_category":"DREDGE","section":"all"} 1`] = `
Object {
  "moreCount": 0,
  "paramsInfo": Array [
    Object {
      "key": "price",
      "label": "Цена",
      "val": "1 — 2 ₽",
    },
    Object {
      "key": "dredge_type",
      "label": "Тип экскаватора",
      "val": "Гусеничный экскаватор, Траншейный экскаватор",
    },
  ],
}
`;
